### PR TITLE
Testing | Cyril/add bundling to build process

### PR DESCRIPTION
**Pull request by cyrielo**
# Description
This PR remove react_rails dependency and uses webpack to bundle the entire react application with npm as the new dependency manager.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix a feature that knowingly causes existing functionality to not work as expected)

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?

- [x] Login Screen
- [x] Single Upload Page
- [x] Batch Upload Page
- [x] All Projects Page
- [x] Sample Details Page
- [x] Report Page

Specific components 

* 

# Testing Script:

### Login Page
* Successfully login to the application and get taken to the project page

### Single Upload
* Using an AWS path that doesn't exist throws an error
* Successfully uploaded files from a folder (use test host)
* Uploading a duplicate file throws an error
* Pipeline gets initiated successfully 

### Batch Upload
* Using an AWS path that doesn't exist throws an error
* Successfully uploaded files from a folder (use test host)
* Navigate to batch upload settings page after upload 
* Host selection from the previous page still selected 
* Uploading a duplicate file throws a warning
* Pipeline gets initiated successfully 

### All Project Page
* Switch between projects 
* Search for a sample
* Navigate through different pages of samples
* Select Upload new sample button
* Select sample to see generated report

### Sample Details Page
* Information is consistent from Project Page
* Notes are editable and save successfully 
* Can navigate to report page 

### Report Page
* Data loads successfully 
* Collapse and Uncollapse button works
* Clicking on the taxonomy name gives you total reads for that taxon
* Category filter works
* Genus Search works
* Filters work as expected
* Disabling Filters works
* Filtering by column works 

# Checklist:

- [x] I have run the testing script to make sure current functionality is unchanged
- [x] I have done relevant tests that prove my fix is effective or that my feature works
- [x] I have spent time testing out edge cases for my feature
- [x] I have updated the test script or pull request template if necessary
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by cyrielo** [comment]
yeah, materialize uses different fonts when we want bold text or lighter text. Also, fontawesome fonts are included too. I think we can be able to inline all the fonts, I would look into that.

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -13,4 +13,10 @@ def hash_array_json2csv(input_file, output_file, keys)
       end
     end
   end
+
+  def escape_json(hash)
+    str = hash.to_json unless hash.class == 'String'
+    str = str.gsub!("\\", "\\\\\\") if str.include? "\\"
+    str
+  end
```
Comment (cdebourcy) [inline app/helpers/application_helper.rb]: nice, as we discussed. Upon further reflection, there must be a built-in, more secure way to pass the data payload to react_component() without this type of manual escape. I've pinged Trevor for help, hopefully we can discuss soon

Comment (cdebourcy) [reply app/helpers/application_helper.rb]: Also @yunfangjuan ?

**Comment by cyrielo** [comment]
This PR was meant to test the changes, and give an idea of the changeset necessary for the migration to the entire team. This PR is no longer relevant as there's another PR for the actual work
